qemu/tests/cfg/dump_guest_memory.cfg

```diff
@@ -0,0 +1,37 @@
+- dump_guest_memory:
+    type = dump_guest_memory
+    virt_test_type = qemu
+    no Windows
+    monitors = 'qmp1'
+    monitor_type_qmp1 = qmp
+    x86_64:
+        extra_params = "-device vmcoreinfo"
+    variants:
+        - with_detach_params:
+            qmp_cmd = "dump-guest-memory detach=true, paging=false, protocol=file:/home/dump"
+            qmp_cmd += "$query-dump"
+            cmd_result_check = contain
+            cmd_return_value = "{}"
+            cmd_return_value += "${'status': 'active'}"
+        - verify_diff_format_dump_file:
+            qmp_cmd = "dump-guest-memory paging=false, protocol=file:/home/dump"
+            qmp_cmd += "$query-dump-guest-memory-capability"
+            qmp_cmd += "$dump-guest-memory paging=false, protocol=file:/home/dump, format=quux"
+            qmp_cmd += "$dump-guest-memory paging=false, protocol=file:/home/dump, format=elf"
+            qmp_cmd += "$dump-guest-memory paging=false, protocol=file:/home/dump, format=kdump-zlib"
+            qmp_cmd += "$dump-guest-memory paging=false, protocol=file:/home/dump, format=kdump-lzo"
+            qmp_cmd += "$dump-guest-memory paging=false, protocol=file:/home/dump, format=kdump-snappy"
+            qmp_cmd += "$dump-guest-memory paging=true, protocol=file:/home/dump, format=kdump-snappy"
+            qmp_cmd += "$dump-guest-memory paging=true, protocol=file:/home/dump, format=kdump-zlib"
+            qmp_cmd += "$dump-guest-memory paging=true, protocol=file:/home/dump, format=kdump-lzo"
+            cmd_result_check = contain
+            cmd_return_value = "{}"
+            cmd_return_value += "${'formats': ["elf", "kdump-zlib", "kdump-lzo", "kdump-snappy"]}"
+            cmd_return_value += "${'desc': "Invalid parameter 'quux'"}"
+            cmd_return_value += "${}"
+            cmd_return_value += "${}"
+            cmd_return_value += "${}"
+            cmd_return_value += "${}"
+            cmd_return_value += "${"desc": "kdump-compressed format doesn't support paging or filter"}"
+            cmd_return_value += "${"desc": "kdump-compressed format doesn't support paging or filter"}"
+            cmd_return_value += "${"desc": "kdump-compressed format doesn't support paging or filter"}"
```

Comment:
Honestly this looks a mess for me, I would recommend to make the command and the result be a pair to make the code be easier to management.
Either you have different variants for different format, like:
```
- verify_diff_format_dump_file:
    cmd_result_check = contain
    dump_memory_cmd = "dump-guest-memory paging=false, protocol=file:/home/dump"
    variants:
        - @ default:
            qmp_cmd = ${dump_memory_cmd}
            cmd_return_value = "{}"
        - quux:
            qmp_cmd += "${dump_memory_cmd}, format=quux"
            cmd_return_value += "${'desc': "Invalid parameter 'quux'"}"
        ...
```
Or separately define commands with different name and then check them one by one in a loop. like:
```
cmd_list = 'default_dump dump_capability format_quux ...'
default_dump_qmp_cmd = "dump-guest-memory paging=false, protocol=file:/home/dump"
dump_capability_qmp_cmd = "$query-dump-guest-memory-capability"
format_quux_qmp_cmd = "${default_dump_qmp_cmd}, format=quux"
...
default_dump_return_value = "{}"
...
```
```
    cmd_list = params.objects('cmd_list')
    for cmd in cmd_list:
        output = qmp_port.send_args_cmd(params['%s_qmp_cmd' % cmd])
        ...
        check_result(output, params['%s_return_value' % cmd])
```

Reply:
Modified the code as you said, could you please review this patch again?
Thanks!